pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,327 @@
+        """
+        self.con = con
+
+    def get(
+        self,
+        namespace: str = None,
+        name: str = None,
+        tag: str = None,
+        query: str = None,
+        admin: Union[List[str], str] = None,
+        limit: int = DEFAULT_LIMIT,
+        offset: int = DEFAULT_OFFSET,
+    ) -> AnnotationList:
+        """
+        Get project annotations.
+        There is 5 scenarios how to get project or projects annotations:
+            - provide name, namespace and tag. Return: project annotations of exact provided PK(namespace, name, tag)
+            - provide only namespace. Return: list of projects annotations in specified namespace
+            - Nothing is provided. Return: list of projects annotations in all database
+            - provide query. Return: list of projects annotations find in database that have query pattern.
+            - provide query and namespace. Return:  list of projects annotations find in specific namespace
+                that have query pattern.
+        :param namespace: Namespace
+        :param name: Project name
+        :param tag: tag
+        :param query: query (search string): Pattern of name, tag or description
+        :param admin: admin name (namespace), or list of namespaces, where user is admin
+        :param limit: return limit
+        :param offset: return offset
+        :return: pydantic model: AnnotationReturnModel
+        """
+        if all([namespace, name, tag]):
+            found_annotation = [
+                self._get_single_annotation(
+                    namespace=namespace,
+                    name=name,
+                    tag=tag,
+                    admin=admin,
+                )
+            ]
+            return AnnotationList(
+                count=len(found_annotation),
+                limit=1,
+                offset=0,
+                results=found_annotation,
+            )
+        return AnnotationList(
+            limit=limit,
+            offset=offset,
+            count=self._count_projects(
+                namespace=namespace, search_str=query, admin=admin
+            ),
+            results=self._get_projects(
+                namespace=namespace,
+                search_str=query,
+                admin=admin,
+                offset=offset,
+                limit=limit,
+            ),
+        )
+
+    def get_by_rp(
+        self,
+        registry_paths: Union[List[str], str],
+        admin: Union[List[str], str] = None,
+    ) -> AnnotationList:
+        """
+        Get project annotations by providing registry_path or list of registry paths.
+        :param registry_paths: registry path string or list of registry paths
+        :param admin: list of namespaces where user is admin
+                :return: pydantic model: AnnotationReturnModel(
+            limit:
+            offset:
+            count:
+            result: List [AnnotationModel])
```

Comment:
While integrating with pephub, sometimes I will see `result` and sometimes I see `results` (plural). I was wondering if we could standardize that

Reply:
I did it already 😀

Reply:
Oh good. thats probably why I couldnt find a concrete example lol